### PR TITLE
4.x : Refactor cucumber tests

### PR DESCRIPTION
**Pull request by deepakrkris**
- [ ] - Abstract test functions in cucumber world and provide reusable test functions
- [ ] - move `Find` test cases to separate .js files
- [ ] - remove the functionality to create sandbox dir with TYPE_OF_TEST
- [ ] - remove variable declaration TYPE_OF_TEST = 'acceptance'

### CODE REVIEW

File: test/acceptance/use-cases/test-scripts/test-setup.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  this.registerHandler('BeforeFeature', function(event, next) {
+    return next();
+  });
+  this.registerHandler('AfterFeature', function(event, next) {
+    return next();
+  });
+};
+
+const WorldConstructor = function(callback) {
+  const world = require('../../../helpers/test-suite.js');
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/test-setup.js]: remove .js

File: test/acceptance/use-cases/test-scripts/test-setup.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const workspaceManager = require('../../../../component/workspace-manager');
+
+module.exports = function() {
+  this.World = WorldConstructor;
+  this.registerHandler('BeforeFeature', function(event, next) {
+    return next();
+  });
+  this.registerHandler('AfterFeature', function(event, next) {
+    return next();
+  });
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/test-setup.js]: nice, remove

File: test/acceptance/use-cases/test-scripts/test-setup.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const testSupport = require('../../../helpers/test-support');
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/test-setup.js]: remove

File: test/acceptance/use-cases/test-scripts/test-setup.js (hunk 1 above)
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/test-setup.js]: move to top

File: test/acceptance/use-cases/test-scripts/test-setup.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+const workspaceManager = require('../../../../component/workspace-manager');
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/test-setup.js]: remove

File: test/acceptance/use-cases/test-scripts/create-datasource.js [hunk 5]
```diff
@@ -19,86 +18,27 @@ module.exports = function() {
...
-    testsuite.workspace =
-      workspaceManager.getWorkspaceByFolder(testsuite.workspaceDir);
-    testsuite.workspaceId = testsuite.workspace.getId();
+    testsuite.templateName = templateName;
     next();
   });
 
   this.When(/^I create datasource '(.+)' with connector '(.+)'$/,
   function(dsName, connector, next) {
-    testsuite.datasourceId = 'common.datasources.' + dsName;
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/create-datasource.js]: connectorName

File: test/acceptance/use-cases/test-scripts/create-datasource.js [hunk 6]
```diff
@@ -19,86 +18,27 @@ module.exports = function() {
...
+    this.setup(testsuite.templateName);
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/create-datasource.js]: move up

File: test/acceptance/use-cases/test-scripts/find-datasource.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
+'use strict';
+const app = require('../../../../');
+const expect = require('../../../helpers/expect');
+const fs = require('fs-extra');
+const loopback = require('loopback');
+const path = require('path');
+const testSupport = require('../../../helpers/test-support');
+const util = require('util');
+const workspaceManager = require('../../../../component/workspace-manager');
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/find-datasource.js]: move to world

File: test/acceptance/use-cases/test-scripts/find-datasource.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+const DataSourceDefinition = app.models.DataSourceDefinition;
+
+app.on('booted', function() {
+  app.emit('ready');
+});
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/find-datasource.js]: move to world

File: test/acceptance/use-cases/test-scripts/find-datasource.js [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+const app = require('../../../../');
+const expect = require('../../../helpers/expect');
+const fs = require('fs-extra');
+const loopback = require('loopback');
+const path = require('path');
+const testSupport = require('../../../helpers/test-support');
+const util = require('util');
+const workspaceManager = require('../../../../component/workspace-manager');
+
+const DataSourceDefinition = app.models.DataSourceDefinition;
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/find-datasource.js]: move into test body

File: test/acceptance/use-cases/test-scripts/find-datasource.js [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+app.on('booted', function() {
+  app.emit('ready');
+});
+
+module.exports = function() {
+  const testsuite = this;
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/find-datasource.js]: let workspace;

File: test/acceptance/use-cases/test-scripts/find-datasource.js [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+  this.When(/^I query for datasource '(.+)' from workspace '(.+)'$/,
+  function(dsName, workspaceName, next) {
+    const workspaceId = testsuite.workspace.getId();
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/find-datasource.js]: workspaceId = workspace.getId();

File: test/acceptance/use-cases/test-scripts/find-datasource.js [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+    testsuite.datasourceId = dsName;
+    const filter = {where: {id: testsuite.datasourceId}};
+    DataSourceDefinition.find(
+      filter,
+      options,
+      function(err, data) {
+        if (err) return next(err);
+        testsuite.datasource = data;
+        next();
+      });
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/find-datasource.js]: move to world

File: test/acceptance/use-cases/test-scripts/find-datasource.js [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+  });
+
+  this.Then(/^the datasource definition is returned$/, function(next) {
+    expect(Object.keys(testsuite.datasource))
+    .to.include.members(['connector', 'name']);
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/find-datasource.js]: indent

File: test/acceptance/use-cases/test-scripts/find-datasource.js [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+    next();
+  });
+
+  this.When(/^I update datasource '(.+)' with connector '(.+)'$/,
+  function(dsName, connector, next) {
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/find-datasource.js]: datasource, connector or dsName, connectorName

File: test/acceptance/use-cases/test-scripts/find-datasource.js [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+      });
+  });
+
+  this.Then(/^the datasource definition is returned$/, function(next) {
+    expect(Object.keys(testsuite.datasource))
+    .to.include.members(['connector', 'name']);
+    next();
+  });
+
+  this.When(/^I update datasource '(.+)' with connector '(.+)'$/,
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/find-datasource.js]: move to update script instead

File: test/acceptance/use-cases/test-scripts/create-from-templates.js [hunk 16]
```diff
@@ -8,83 +8,67 @@ const path = require('path');
...
+    this.createWorkspace(Workspace, templateName, next);
   });
 
   this.Then(/^the workspace is created$/, function(next) {
-    const workspace =
-      workspaceManager.getWorkspaceByFolder(testsuite.destinationPath);
-    const dir = workspace.getDirectory();
-    const template = workspaceManager.getTemplate(testsuite.templateName);
+    const testsuite = this;
+    this.setup(this.templateName);
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/create-from-templates.js]: loadWorkspace

File: test/acceptance/use-cases/test-scripts/create-from-templates.js [hunk 17]
```diff
@@ -8,83 +8,67 @@ const path = require('path');
...
-            const expectedList = [];
-            template.server.datasources.forEach(function(ds) {
-              expectedList.push(ds.name);
-            });
-            expect(Object.keys(datasources)).to.include.members(expectedList);
-            cb();
-          });
-        });
-      }
+      addTaskToCheckFacet(tasks, testsuite);
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/create-from-templates.js]: verify

File: test/acceptance/use-cases/test-scripts/create-from-templates.js [hunk 18]
```diff
@@ -8,83 +8,67 @@ const path = require('path');
...
+    }
+    if (template.server.modelConfigs) {
+      addTaskToCheckModelConfig(tasks, testsuite, template);
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/create-from-templates.js]: ditto

File: test/acceptance/use-cases/test-scripts/create-from-templates.js [hunk 19]
```diff
@@ -8,83 +8,67 @@ const path = require('path');
...
+    };
+    if (template.server.datasources) {
+      addTaskToCheckDataSource(tasks, testsuite, template);
```
Comment (superkhau) [inline test/acceptance/use-cases/test-scripts/create-from-templates.js]: ditto

**Comment by deepakrkris** [comment]
@slnode test please

**Pull request merged**